text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1417 @@
+> different protocols and configurations should not vary between a Rust and Go implementation. However, the measurement
+> times might be a bit different with a Go implementation. A Go implementation is under development, the results will be
+> integrated in this document as soon as possible.
+
+### Metrics
+
+#### Batch Size Analysis
+
+In the dataset used, each measurement point is composed of a timestamp, 9 metrics sharing 10 attributes. The dataset
+contains 200K measurement points. To transmit this dataset with OTLP, we must first convert this dataset into a stream
+of univariate
+metrics (i.e. one metric per data point). The special attribute 'state' is added to specify the nature of the metric.
+This format follows the same convention as the cpu or memory system metrics issued by the OTEL collector.
+
+The following multi-line plot shows for different batch sizes, representation modes (OTLP or OTLP Arrow) and compression
+algorithms the batch size in bytes once compressed. In this test, the data are not transformed into multivariate
+time-series for OTLP Arrow. This will be the subject of the next plot.
+
+![Univariate metrics bytes](img/0156_univariate_metrics_bytes.png)
+
+3 compression algorithms have been evaluated for OTLP batches (Zlib, Lz4, and Zstd) in order to compare their efficiency
+on a "row-oriented" representation. According to our results Zlib and Zstd are the two best compression algorithms for
+OTLP data with a very slight advantage for Zlib. The same compression algorithms have been tested on our new
+"column-oriented" representation. Zlib and Zstd are the two best compression algorithms for OTLP Arrow.
+
+> As this plot shows, the columnar representation is almost twice as efficient in terms of bandwidth gain. However, it
+> is
+> possible to do much better by taking advantage of the native support for multivariate time-series in OTLP Arrow.
+
+The following graph compares the best OTLP compression algorithm (Zlib) with OTLP Arrow + Zstd + multivariate time-series
+support.
+
+![Multivariate metrics bytes](img/0156_multivariate_metrics_bytes.png)
+
+> With this optimization, OTLP Arrow is a little over 4 times more efficient than OTLP. The zstd compression algorithm
+> stands out for columnar data.
+
+Even for small batches of metrics, the OTLP Arrow representation is more efficient than the OTLP representation in a
+*multivariate time-series context* (see table below).
+
+![Metrics small batches](img/0156_metrics_small_batches.png)
+
+#### Serialization/Compression Time Analysis
+
+This section compares the time spent creating, serializing, compressing, decompressing and deserializing batches of
+metrics with OTLP and OTLP Arrow while looking to optimize message size (Zlib for OTLP and Zstd for OTLP Arrow). In both
+cases, the data producer delivers the metrics in a row-oriented approach. The Arrow OTLP implementation therefore
+integrate the
+row-to-column conversion time.
```

Comment:
I can hardly see any time other than "Batch creation" on the charts for Arrow. Is the chart correct? Where is the conversion time reflected?

Reply:
The OTLP -> "OTLP Arrow" conversion time is part of the creation phase in "OTLP Arrow". What is missing is the conversion "OTLP Arrow" -> OTLP when the receiver is not already "OTLP Arrow" compatible. For phase 1 of the collector, this second conversion will be necessary but not for phase 2 or any backend that could be "OTLP Arrow" compatible even before phase 2 of this project. 
I will make adjustments in the specification to clarify these points.